File: src/main/kotlin/aba/simulation/MyMessage.java
package aba.simulation;

import OSPABA.*;

public class MyMessage extends MessageForm
{
	public MyMessage(Simulation sim)
	{
		super(sim);
	}

	public MyMessage(MyMessage original)
	{
		super(original);
		// copy() is called in superclass
	}

	@Override
	public MessageForm createCopy()
	{
		return new MyMessage(this);
	}

	@Override
	protected void copy(MessageForm message)
	{
		super.copy(message);
		MyMessage original = (MyMessage)message;
		// Copy attributes
	}
}

File: src/main/kotlin/aba/simulation/MyMessage.kt
package aba.simulation

import OSPABA.*

class MyMessage : MessageForm {
    constructor(sim: Simulation) : super(sim) {}

    constructor(original: MyMessage) : super(original) {
        // copy() is called in superclass
    }

    override fun createCopy(): MessageForm {
        return MyMessage(this)
    }

    override fun copy(message: MessageForm) {
        super.copy(message)
        val original = message as MyMessage
        // Copy attributes
    }
}
